File: scripts/deploy_contract.py
from web3 import Web3

w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:8545"))

if not w3.is_connected():
    raise Exception("Web3 not connected!")

with open("./contracts/MyNFT.abi", "r") as abi_file:
    abi = abi_file.read()

with open("./contracts/MyNFT.bin", "r") as bin_file:
    bytecode = bin_file.read()

chain_id = 1337
address = w3.eth.accounts[0] # Geth dev account
contract = w3.eth.contract(abi=abi, bytecode=bytecode)

transaction = contract.constructor().build_transaction({
    "from": address,
    "nonce": w3.eth.get_transaction_count(address),
    "gas": 5000000,
    "gasPrice": w3.eth.gas_price,
    "chainId": chain_id
})

# signed_tx = w3.eth.account.sign_transaction(transaction, private_key=private_key)
tx_hash = w3.eth.send_transaction(transaction=transaction)

receipt = w3.eth.wait_for_transaction_receipt(tx_hash)
print(f"Contract deployed at: {receipt.contractAddress}")

File: scripts/deploy_contract.js
const { Web3 } = require('web3');
const fs = require('fs');

const web3 = new Web3('http://127.0.0.1:8545');
const { eth } = web3;

const abi = JSON.parse(fs.readFileSync('./contracts/MyNFT.abi', 'utf8'));
const bytecode = fs.readFileSync('./contracts/MyNFT.bin', 'utf8');

(async () => {
    const accounts = await eth.getAccounts();
    const account = accounts[0]; // Geth dev account

    console.info('Deploying contract')
    const contract = new eth.Contract(abi);
    
    const result = await contract
        .deploy({ data: bytecode })
        .send({
            from: account,
            gas: 5000000,
            gasPrice: web3.utils.toWei('20', 'gwei'),
        });

    console.info('Contract deployed to:', result.options.address);
})()
